groovy-language-server/src/main/java/com/palantir/ls/util/DefaultDiagnosticBuilder.java

```diff
@@ -16,53 +16,13 @@
 
 package com.palantir.ls.util;
 
-import com.google.common.base.Optional;
-import com.google.common.base.Preconditions;
-import io.typefox.lsapi.Diagnostic;
-import io.typefox.lsapi.DiagnosticSeverity;
-import io.typefox.lsapi.Range;
 import io.typefox.lsapi.builders.DiagnosticBuilder;
 
-public final class DefaultDiagnosticBuilder {
+public final class DefaultDiagnosticBuilder extends DiagnosticBuilder {
 
-    private final String message;
-    private final DiagnosticSeverity severity;
-
-    private Optional<String> code = Optional.absent();
-    private Optional<Range> range = Optional.absent();
-    private Optional<String> source = Optional.absent();
-
-    public DefaultDiagnosticBuilder(String message, DiagnosticSeverity warning) {
```

Comment:
The reason for the constructor with a message and warning is to disallow creation of diagnostics without the fields required by the protocol. All the other fields have defaults, or in the case of `code` is optional.